helpers/stream.go

```diff
@@ -0,0 +1,56 @@
+package helpers
```

Comment:
OT: these should _really_ go away. We need to fix the stream interfaces and:

1. Make `Close()` close.
2. Add `CloseWrite()`, `CloseRead()`.

Reply:
Since both funcs take a stream as their first arg, I wonder if we can preserve backwards compatibility by hoisting the instance method to a package var. Will try it out, otherwise I suggest punting it to iteration 3 (backwards-incompatible ergonomics changes).

Reply:
I agree.

Reply:
We might call the `FullyClose` as `Shutdown`.

Reply:
I'll open an issue for this.